web/Controllers/AdminController.cs

```diff
@@ -89,4 +89,58 @@ public async Task<IActionResult> LandlordList()
         landlordListModel.UnapprovedLandlords = await _adminService.GetUnapprovedLandlords();
         return View(landlordListModel);
     }
+
+    [Authorize(Roles = "Admin")]
+    [HttpPost]
+    public Task<ActionResult> GetInviteLink(int landlordId)

```

Comment:
Methods that return `Task` should almost always be async. There's a pretty good summary of dos and don'ts here: https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md and I'm happy to chat through async at some point if it doesn't make sense.

Reply:
yeah sure I'll look over the document
Rider doesn't seem to be very happy when I change it to async as there is no await calls inside the function
Should I change some of the functions in the service to async too and await for them in the controller?

Reply:
Rider doesn't seem to be very happy when I just add async, since there is no await inside the call.

Instead I changed it from Task<ActionResult> to just ActionResult and it seems to be happy.

Is it ok to just do that?

Reply:
Yep, just having "normal" non-async code is fine, we haven't really standardised on one or the other.